pkg/alertmanager/merger/merger.go

```diff
@@ -0,0 +1,15 @@
+package merger
```

Comment:
Super-Nit: Not sure this package carries its own weight. I know there will be v2 implementation too, but it's still looks pretty thin. Same goes for `Merger` interface.

Reply:
I agree and wasn't 100% sure what to do. The `pkg/alertmanager` package is getting very busy, was my reasoning here.

Would a `pkg/alertmanager/distributor` perhaps be better?

Also regarding the interface, I'd be equally happy with a functions, if that's what you meant? (I was mainly taking inspiration from here https://github.com/cortexproject/cortex/blob/94b0137e7133652f32a6ca641510b8c4c861791e/pkg/querier/queryrange/query_range.go#L67)

Reply:
Personally I would just use `pkg/alertmanager`, I think it's still a small package.

> Also regarding the interface, I'd be equally happy with functions, if that's what you meant?

Yes, I was thinking about just using functions here. My reasoning is that 1) we're only going to have 2 implementations (v1, v2 api) and 2) client doesn't need to supply its own implementation. But I am first to admit this is just a personal preference for this specific case, and don't mind keeping it as-is.

Reply:
I don't have a strong opinion here. I personally did like the `merger` package, given it's content is self contained but moving to `pkg/alertmanager` looks good to me as well. Let's just not block on this. V2 merging PR is awaiting for this PR! 😉 

> Would a pkg/alertmanager/distributor perhaps be better?

I wouldn't name the package distributor because we already have `pkg/distributor` and in the past (when we tried `pkg/alertmanager/distributor` at some point) we realised it was causing much confusion when reading the code.

Reply:
Leaving as-is to unblock, happy to refactor after the fact.